scripts/presubmit.sh

```diff
@@ -1,13 +1,19 @@
 #!/bin/bash
 set -ex
 
+regenerate() {
+    go generate -run="protoc" ./...
+    go generate -run="mockgen" ./...
+}
+trap regenerate EXIT
+
 find . -name "*.pb.go" -delete
 find . -name "mock_*" -delete
 golint --set_exit_status ./...
 go vet ./...
 misspell . -locale US
-go generate -run="protoc" ./...
-go generate -run="mockgen" ./...
+regenerate
```

Comment:
The benefit is that if you happen to have a golint error when running locally, then `presubmit` doesn't leave you with an uncompilable codebase (because all the generated files are missing).